### PR TITLE
PinZheng All Agenda Revamp

### PR DESCRIPTION
**Pull request by Ducksss**
# Description
This is the all agenda page revamp. It allows users to toggle between multiple days as well as categories. For example, users can choose to view events on Day 1 that belongs to Data Arcade or Data Engineering Category. When toggling between the different days, the filters persists and do not reset automatically.

## State persistence filter video
https://user-images.githubusercontent.com/58126222/174995831-abefa0e2-2bbb-441c-89a6-c6bd8771839d.mov

## A detailed view of events
Screenshot 2022-06-22 at 5 33 18 PM

## A detailed view of events that fall within the same timing (being aggregated and grouped) 
Screenshot 2022-06-22 at 5 33 59 PM

## Notable file changes
- _data/events-nav.yml (Adding a new sub section)
- _data/stackx2022dsc.yml (Modifying the dataset by introducing a new variable, timeslot_metadata)
- apps/src/pages/AllSessionsApp.vue (Vue app for the all agenda section)
- apps/src/pages/vue-app-all-sessions.js (Vue Injector)
- assets/icons/.. (Icons)
- collections/_communities/events/conferences/stack-x-data-science-connect-2022/agenda/all-agenda.html 
- collections/_communities/events/conferences/stack-x-data-science-connect-2022/agenda/day-1-programme.html (Accounted for dataset modification)
- collections/_communities/events/conferences/stack-x-data-science-connect-2022/agenda/day-2-programme.html (Accounted for dataset modification)
- search/dataScienceConnect.json (Data file used to render the data)

## Notable concerns before going to prod
- Day 2 events in the stackx2022dsc.yaml file, entire demo category is using filler data. As such, we need to consult @candyyap-gvt for more information if this page is going live
- The download Full Agenda PDF has no link at the moment.

### CODE REVIEW

**Review by chewyixin99 — COMMENTED** [review]
1. styling not really optimized for mobile view, pillList does not show all pills in mobile
image

**Comment by Ducksss** [comment]
> 1. styling not really optimized for mobile view, pillList does not show all pills in mobile
>    image

resolved

**Comment by teochengyong** [comment]
@Ducksss the filter day-2 keynote should not be on day 1

image

**Comment by Ducksss** [comment]
> @Ducksss the filter day-2 keynote should not be on day 1
> 
> image

Yep, the metadata_timeslot variable that I have created and populated was wrong. Have updated it from 2022-12-07T13:30:00+08:00 to 2022-12-08T13:30:00+08:00.

**Comment by teochengyong** [comment]
@Ducksss maybe can make the download full agenda pdf optional if there is no such file
image

File: apps/src/pages/AllSessionsApp.vue [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+              </span>
+            </span>
+          </label>
+        </template>
+      </div>
+    </div>
+    <!-- Clear all Filter -->
+    <div>
+      <div class="margin--top--sm">
+        <a
```
[the remark this answers is not in the document]
Comment (Ducksss) [reply apps/src/pages/AllSessionsApp.vue:73]: Woah, did not realise anchor tags and buttons have their own semantic meaning which is interpreted by browsers, screen readers, and SEO crawlers differently... very interesting hahaha...

File: apps/src/pages/AllSessionsApp.vue [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+      categoryOptions,
+      pillSelectedValues,
+      pillCheckboxOptions,
+      clearAllFilters,
+    };
+  },
+};
+</script>
+
+<style scoped>
```
[the remark this answers is not in the document]
Comment (Ducksss) [reply apps/src/pages/AllSessionsApp.vue:322]: Got it 👍

**Comment by Ducksss** [comment]
Note: Commit ade7955 include implementations suggested in this PR as well as in the Dev Portal Telegram Group Chat.

Implementations of suggestions from telegram are:
- pillbox values are sorted through time to be in a chronological order

**Comment by Ducksss** [comment]
^ Implemented some small suggestions suggested by ZiKai, such as adding some padding to the initial unselected pillboxes to account for the ticks when the pillboxes are checked

**Review by chewyixin99 — APPROVED** [review]

**Pull request merged**